### PR TITLE
collations: upgrade to MySQL 8.0.32

### PR DESCRIPTION
**Pull request by vmg**
## Description

Here are some fresh changes to the collations package to more closely match newer versions of MySQL.

Includes:

- New aliases with `utf8mb3_` instead of `utf8_`, in preparation for the eventual switch so `utf8` means `utf8mb4`.
- New `uca900` collations for some weeeeeird languages. We support those now too!
- Updated the `collgen` tooling to support newer versions of MySQL.
- Updated the way collations are stored statically in the binary to use an array instead of a map.

## Related Issue(s)

- Fixes https://github.com/vitessio/vitess/issues/12564

## Checklist

-   [ ] "Backport to:" labels have been added if this change should be back-ported
-   [ ] Tests were added or are not required
-   [ ] Did the new or modified tests pass consistently locally and on the CI
-   [ ] Documentation was added or is not required

### CODE REVIEW

**Comment by dbussink** [comment]
@vmg It looks like there's some `schemadiff` failures here, probably because of changes around the `utf8` vs `utf8mb3` naming.

**Review by dbussink — APPROVED** [review]

File: go/mysql/collations/mysqldata.go [hunk 1]
```diff
@@ -119,7 +119,7 @@ var tounicode_dec8_swedish_ci = [...]uint16{
...
...x61, 0x62, 0x63, 0x64, 0x65, 0x66, 0x67, 0x68, 0x69, 0x6a, 0x6b, 0x6c, 0x6d, 0x6e, 0x6f, 0x70, 0x71, 0x72, 0x73, 0x74, 0x75, 0x76, 0x77, 0x78, 0x79, 0x7a, 0x7b, 0x7c, 0x7d, 0x7e, 0x7f, 0x80, 0x81, 0x82, 0x83, 0x84, 0x85, 0x86, 0x87, 0x88, 0x89, 0x8a, 0x8b, 0x8c, 0x8d, 0x8e, 0x8f, 0x90, 0x91, 0x92, 0x93, 0x94, 0x95, 0x96, 0x97, 0x98, 0x99, 0x9a, 0x9b, 0x9c, 0x9d, 0x9e, 0x9f, 0xa0, 0xa1, 0xa2, 0xa3, 0xa8, 0xa5, 0x0, 0xa7, 0x0, 0xa9, 0xaa, 0xab, 0x0, 0x0, 0x0, 0x0, 0xb0, 0xb1, 0xb2, 0xb3, 0x0, 0xb5, 0xb6, 0xb7, 0x0, 0xb9, 0xba, 0xbb, 0xbc, 0xbd, 0x0, 0xbf, 0xc0, 0xc1, 0xc2, 0xc3, 0xc4, 0xc5, 0xc6, 0xc7, 0xc8, 0xc9, 0xca, 0xcb, 0xcc, 0xcd, 0xce, 0xcf, 0x0, 0xd1, 0xd2, 0xd3, 0xd4, 0xd5, 0xd6, 0x0, 0xd8, 0xd9, 0xda, 0xdb, 0xdc, 0x0, 0x0, 0xdf, 0xe0, 0xe1, 0xe2, 0xe3, 0xe4, 0xe5, 0xe6, 0xe7, 0xe8, 0xe9, 0xea, 0xeb, 0xec, 0xed, 0xee, 0xef, 0x0, 0xf1, 0xf2, 0xf3, 0xf4, 0xf5, 0xf6, 0x0, 0xf8, 0xf9, 0xfa, 0xfb, 0xfc, 0x0, 0x0, 0xfd}}, eightbit.UnicodeMapping{From: 0x152, To: 0x178, Range: []uint8{0xd7, 0xf7, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xdd}}}
```
Comment (systay) [inline go/mysql/collations/mysqldata.go:122]: just asking - it wouldn't be easier to embed these as uuencoded files, right?

**Review by systay — APPROVED** [review]
LGTM

File: go/mysql/collations/mysqldata.go (hunk 1 above)
Comment (vmg) [reply go/mysql/collations/mysqldata.go:122]: No, the raw tables are embedded, but the 8bit mappings are slightly more complex data structures, so they cannot be loaded directly from an embedding.

**Review by frouioui — APPROVED** [review]
I think we also need to update the website documentation too (https://vitess.io/docs/17.0/user-guides/configuration-basic/collations/).

Otherwsie, LGTM :)

**Comment by vmg** [comment]
Ah yes, I need to open a docs PR for this. Thanks for reminding me. :+1:

**Pull request merged**